server/plugin/api.go

```diff
@@ -1035,70 +959,223 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+		c.Log.WithError(err).Warnf("Error decoding the JSON body")
+		p.writeAPIError(w, &serializer.APIErrorResponse{ID: "", Message: "Please provide a valid JSON object.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	githubClient := p.githubConnectUser(c.Context.Ctx, c.GHInfo)
+	issue, _, err := githubClient.Issues.Get(c.Ctx, req.RepoOwner, req.RepoName, req.IssueNumber)
 	if err != nil {
-		p.writeAPIError(w, &APIErrorResponse{Message: "Invalid param 'number'.", StatusCode: http.StatusBadRequest})
+		// If the issue is not found, it probably belongs to a private repo.
+		// Return an empty response in that case.
+		var gerr *github.ErrorResponse
+		if errors.As(err, &gerr) && gerr.Response.StatusCode == http.StatusNotFound {
+			c.Log.WithError(err).With(logger.LogContext{
+				"owner":  req.RepoOwner,
+				"repo":   req.RepoName,
+				"number": req.IssueNumber,
+			}).Debugf("Issue not found")
+			p.writeJSON(w, nil)
+			return
+		}
+
+		c.Log.WithError(err).With(logger.LogContext{
+			"owner":  req.RepoOwner,
+			"repo":   req.RepoName,
+			"number": req.IssueNumber,
+		}).Debugf("Could not get the issue")
+		p.writeAPIError(w, &serializer.APIErrorResponse{Message: "Could not get the issue", StatusCode: http.StatusInternalServerError})
+		return
+	}
+
+	description := ""
+	if issue.Body != nil {
+		*issue.Body = mdCommentRegex.ReplaceAllString(issue.GetBody(), "")
+		description = *issue.Body
+	}
+
+	assignees := make([]string, len(issue.Assignees))
+	for index, user := range issue.Assignees {
+		assignees[index] = user.GetLogin()
+	}
+
+	labels := make([]string, len(issue.Labels))
+	for index, label := range issue.Labels {
+		labels[index] = label.GetName()
+	}
+
+	milestoneTitle := ""
+	var milestoneNumber int
+	if issue.Milestone != nil && issue.Milestone.Title != nil {
+		milestoneTitle = *issue.Milestone.Title
+		milestoneNumber = *issue.Milestone.Number
+	}
+
+	userID := r.Header.Get(constants.HeaderMattermostUserID)
+	post, appErr := p.API.GetPost(req.PostID)
+	if appErr != nil {
+		p.writeAPIError(w, &serializer.APIErrorResponse{ID: "", Message: fmt.Sprintf("failed to load the post %s", req.PostID), StatusCode: http.StatusInternalServerError})
+		return
+	}
+	if post == nil {
+		p.writeAPIError(w, &serializer.APIErrorResponse{ID: "", Message: fmt.Sprintf("failed to load the post %s : not found", req.PostID), StatusCode: http.StatusNotFound})
+		return
+	}
+
+	p.API.PublishWebSocketEvent(
+		wsEventCreateOrUpdateIssue,
+		map[string]interface{}{
+			"title":            *issue.Title,
+			"channel_id":       post.ChannelId,
+			"postId":           req.PostID,
+			"milestone_title":  milestoneTitle,
+			"milestone_number": milestoneNumber,
+			"assignees":        assignees,
+			"labels":           labels,
+			"description":      description,
+			"repo_full_name":   fmt.Sprintf("%s/%s", req.RepoOwner, req.RepoName),
+			"issue_number":     *issue.Number,
+		},
+		&model.WebsocketBroadcast{UserId: userID},
+	)
```

Comment:
Why is a websocket event used and not a standard HTTP response?

Reply:
Now used HTTP response